manim/utils/hashing.py

```diff
@@ -30,56 +32,122 @@ def default(self, obj):
             Python object that JSON encoder will recognize
 
         """
-        if inspect.isfunction(obj) and not isinstance(obj, ModuleType):
+        if not (isinstance(obj, ModuleType)) and isinstance(
+            obj, (MethodType, FunctionType)
+        ):
             cvars = inspect.getclosurevars(obj)
             cvardict = {**copy.copy(cvars.globals), **copy.copy(cvars.nonlocals)}
             for i in list(cvardict):
                 # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
                 if isinstance(cvardict[i], ModuleType):
                     del cvardict[i]
-            return {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            return self._check_iterable(
+                {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            )
         elif isinstance(obj, np.ndarray):
-            return list(obj)
+            if obj.size > 1000:
+                return f"Np array too big {obj.size}"
```

Comment:
Can we truncate the array instead of just doing this? We could truncate it, and then append the string `"truncated"` to the end of the `repr` and return that instead.

Reply:
Very good idea. We could even hash it.